r-package/grf/vignettes/rate.Rmd

```diff
@@ -0,0 +1,216 @@
+  annotate(geom = 'text', x = 0.15, y = 0.6, color = 'black', hjust = -0.1,
+           label = "RATE")
+```
+
+There are different ways this area can be summed up, which gives rise to different RATE metrics. We refer to the integral above as the "AUTOC". This implies placing more weight on areas under the curve where the expected treatment benefit is largest. When non-zero treatment effects are concentrated among a small subset of the population this weighting has greater power for testing against a sharp null (RATE=0) as upweighting the individuals with a non-zero treatment effect increases the RATE.
+
+Another way to sum this area would be to weight each point on the curve by $q$: $\int_0^1 q \textrm{TOC}(q) dq$. We refer to this metric as the "QINI" (Radcliffe, 2007). This weighting implies placing as much weight on units with low treatment effects as on units with high treatment effects. When non-zero treatment effects are more diffuse across the entire population, this weighting tends to give greater power when testing against a null effect.
+
+Another thing to note is that the "high-vs-low" out-of-bag CATE quantile construction used in Athey and Wager (2019) can also be seen as a kind of RATE, where in place of Qini's identity weighting it would have point mass at q = a given quantile.
+
+## Estimating the RATE
+The overview in the previous section gave a stylized introduction where we imagined we knew $\tau(X_i)$. In practice these have to be estimated and the appropriate *feasible definition* (using $S(X_i) = \hat \tau(X_i)$ as an example) is: the TOC curve ranks all observations on a test set $X^{test}$ according to a CATE function $\hat \tau^{train}(\cdot)$ estimated on a training set, and compares the ATE for the top q-th fraction of units prioritized by $\hat \tau^{train}(X^{test})$ to the overall ATE:
+
+\begin{equation*}
+\begin{split}
+&\textrm{TOC}(q) = E[Y^{test}_i(1) - Y^{test}_i(0) \,|\, \hat \tau^{train}(X^{test}_i) \geq \hat F^{-1}(1 - q)]\\
+& \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \  - E[Y^{test}_i(1) - Y^{test}_i(0)],
+\end{split}
+\end{equation*}
+
+where $\hat F$ is the empirical distribution function of $\hat \tau^{train}(X^{test})$. GRF estimates
+
+$$\tau(X_i)= [Y^{test}_i(1) - Y^{test}_i(0) \,|\, X_i = x]$$
+
+using a suitable doubly robust score construction $\Gamma_i \approx \tau(X_i)$ which is valid in a variety of settings, enabling inference with the bootstrap, and the final AUTOC estimate is constructed from $\frac{1}{n} \sum_{i=1}^{n} \textrm{TOC}(\frac{i}{n})$. This functionality is implemented in the function `rank_average_treatment_effect` which as first argument takes an "evaluation" forest which computes the appropriate score construction $\hat \Gamma_i$ and second argument estimated prioritization scores for units in the evaluation set, using for example a CATE function estimated on a training set (with additional support for forests trained with sample weights or clusters).
```

Comment:
$\Gamma_i \approx \tau(X_i)$ is not generally true, however $\mathbb{E}[ \Gamma_i | X_i] \approx \tau(X_i)$ is.

Reply:
Also we should be careful about $\hat \Gamma_i$ vs $\Gamma_i$. Not sure which you intended to use here, or if you intended to use different ones in the beginning and end of paragraph to indicate the replacement of the true nuisance parameters with estimates. If the latter, it's good to be specific about it in the writing, as it can be confusing if you don't a priori know what it means.

Reply:
I only wanted to convey to the user that given S(X), GRF computes the TOC using using doubly robust scores and not direct tau(X) estimates on the evaluation set, as I can imagine that would be something that could confuse users who end up poking into this, I've added a parenthesis to this paragraph, if you think it's better now?

Reply:
I think ideally there would be a one-liner explaining what a DR score is and why they're used - I did not manage to come up with a good one

Reply:
@erikcs for DR score explanation what about something like the following: 

“where $\hat F$ is the empirical distribution function of $\hat \tau^{train}(X^{test})$. _In order to estimate these expectations, we employ the use of doubly robust scores “scores”, $\hat \Gamma_i$, with the property that they act as nearly unbiased (but noisy) proxies for the true CATE._ GRF estimates [...] using suitable  with the property that they act as nearly unbiased (but noisy) proxies for [...]"

Reply:
But yes I think the parenthetical is helpful here. I might also clarify that the "evaluation" forest is trained on the test set. Though maybe that's already obvious and trying to be explicit here would just be more confusing...

Reply:
And I agree with Steve Y that it's probably worth being at least consistent about our use of $\hat \Gamma$ vs. $\Gamma$. In particular the  $\mathbb{E}[ \Gamma_i | X_i] \approx \tau(X_i)$ should be  $\mathbb{E}[ \hat \Gamma_i | X_i] \approx \tau(X_i)$.

Might also be worth changing $\frac{1}{n} \sum_{i=1}^{n} \textrm{TOC}(\frac{i}{n})$ to $\frac{1}{n} \sum_{i=1}^{n} \hat \textrm{TOC}(\frac{i}{n})$, to maintain consistency for folks who are trying to connect equations in the paper with implementation in `grf`.

Reply:
> 

Yeah that kind of just reads like a tautology given E[Gamma] ~ tau, I was thinking of something along the efficiency explanation of DR scores used in ATE estimation in the first paragraph here https://grf-labs.github.io/grf/REFERENCE.html#average-treatment-effects

Reply:
In the rate paper you write 
```
The approach for replacing τ with Γ has been used for average
treatment effect estimation (Chernozhukov et al., 2018a), conditional average treatment
effect estimation (Kennedy, 2020), policy learning (Athey and Wager, 2021), and structural
functions of treatment effects (Semenova and Chernozhukov, 2017). The key to proving
Theorem 4 (done in Section 8) is extending this technique to L-statistics
```

It would be nice with an easy to digest sentence that gives intuition on WHY this is essential. Then the paragraph in this section could end with something like "You might ask why we replace estimates τ with Γ when computing the TOC when τ are already "valid" point estimates of the CATE. This is the same strategy GRF takes in functions like `best_linear_projection` and `average_treatment_effect`, because this score construction is shown to give semiparametrically efficient and valid inference (see XXX). The intuition is that the CATE point estimates would understate the sampling variation in ... as they rely on estimated "nuisance" components ... <--- I don't know how to explain this well